### ISSUE TITLE
Copy self:provider.tags into all resources

### ISSUE THREAD

**Issue by onnimonni** [serverless/serverless#5593]
# This is a Feature Proposal
**TL;DR:** I would like to use provider tags for all created resources

## Description
AFAIK almost companies use tagging for splitting costs into different departments inside the organisation.

For example if I use `InvoiceID` like this
```yaml
service:
  name: lambda-service-name

provider:
  name: aws
  stage: ${env:ENVIRONMENT, 'development'}
  tags:
    InvoiceID: 1234567

resources:
  Resources:
    S3Reports:
      Type: AWS::S3::Bucket
      Properties:
        BucketName: your-bucket-name-here
```
If I create functions with this configuration they will correctly get the `InvoiceID` tag.
But the `S3::Bucket` included here won't and I need to duplicate this tag into each resources.

It's weird that serverless still adds the `STAGE` tag but not the others:
<img width="390" alt="screen shot 2018-12-13 at 1 51 14 pm" src="https://user-images.githubusercontent.com/5691777/49937034-37f81480-fede-11e8-8896-8a24e1b893de.png">

Could we add all tags into all resources? Or a helper function which I could use to add all tags easily into all resources?

**Comment by c2xbrhdj6u4qs15** [comment]
Hi, I would like to go even a step further and would like to pass a json string to the serverless command to propagate the tags.

`serverless deploy --tags '{"key": "value"}'`

In my serverless.yml I would have this:

`provider:
  name: aws
  tags: ${opt:tags}`

Would this be possible?

Thanks

**Comment by jplock** [comment]
if you use `provider.stackTags` Cloudformation will copy the tags to as many resources that support tags as it can (we have seen though that Cloudfront Distributions do not get the tags however).

**Comment by HarryCaveMan** [comment]
Documentation implies that all resources, including apiGateway stages will receive the `stackTags`, I have so far found that no tags are applied to my api gateway stage, or to the IAM roles created, both of which support tagging.

**Comment by vedant-khandelwal** [comment]
stackTags is also not applied to S3 buckets created in Resources section for some reason

**Comment by shierro** [comment]
I think this feature is very useful. would be happy to see it live in action

**Comment by ethanherbertson** [comment]
> if you use `provider.stackTags` Cloudformation will copy the tags to as many resources that support tags as it can (we have seen though that Cloudfront Distributions do not get the tags however).
> 
> https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-resource-tags.html

Yeah, there has been a pretty significant lag between AWS resource types getting tagging support, and those same resource types getting tagging support *in CloudFormation stack tags*. Not much can be done about that.

I think there are also some resource types that support tagging, but that couldn't be tagged by CloudFormation *at all*. This was certainly the case, at least for a while, for CloudWatch alarms.

In the meantime, Serverless plugins can plug some of the gaps. I just discovered https://github.com/namitkewat/serverless-plugin-tag-events today, for example. Seems like a reasonable pattern.

**Comment by nbcchen** [comment]
doesn't add the tags to s3 buckets created under `provider:s3`
https://www.serverless.com/framework/docs/providers/aws/events/s3
